frontend/src/components/ResponseQuestion/FileUploadResponse/index.tsx

```diff
@@ -87,10 +87,13 @@ const FileUploadResponse: FC<IQuestionResponse<IFileUploadQuestion> & IFileUploa
             // start sending files to the server
             promises.push(uploadFile(file));
         }
+        const result = [];
         (await Promise.all(promises))
             .forEach(res => {
-                setFiles([...files, res]);
+                result.push(res);
+                // setFiles([...files, res]);
             });
+        setFiles(result);
```

Comment:
Fix file upload algorithm. The previous version was wrong because setFiles function is asynchronous.